src/shared/lib/LollipopPlotUtils.ts

```diff
@@ -0,0 +1,60 @@
+import {Mutation} from "shared/api/generated/CBioPortalAPI";
+import {longestCommonStartingSubstring} from "./StringUtils";
+import {countMutationsByProteinChange} from "./MutationUtils";
+import {getTextWidth} from "./wrapText";
+
+export function lollipopLabelText(mutationsAtPosition:Mutation[], size?: number): string {
+    const mutationCountsByProteinChange = countMutationsByProteinChange(mutationsAtPosition);
+
+    // only pick specified number of protein change values
+    const proteinChanges = mutationCountsByProteinChange
+        .map(m => m.proteinChange)
+        .slice(0, size && size > 0 ? size : undefined);
+
+    // sort alphabetically (to make it easier to find longest common starting substring)
+    const proteinChangesSorted = proteinChanges.slice(0).sort();
```

Comment:
didn't you already call slice above?

Reply:
this one is to make another copy to store the sorted array without mutating `proteinChanges`. so `slice()` here is only used for the purpose of cloning, may not be the best way to do it though